File: kids-app/libs/share/src/api/event/event.feedback.dto.ts
import { RatingEnum } from "../../util/enums/rating.enum";

/**
 * @class EventFeedBackDto
 * @description Datenübertragungsobjekt (DTO) zur Repräsentation des Feedbacks für ein Event.
 * Enthält Informationen über den Benutzer, die Bewertung und eine Nachricht.
 */
export class EventFeedBackDto {
    /**
     * @property {string} userId - Die eindeutige Kennung des Benutzers, der das Feedback abgegeben hat.
     */
    userId: string;

    /**
     * @property {string} userName - Der Name des Benutzers, der das Feedback abgegeben hat.
     */
    userName: string;

    /**
     * @property {RatingEnum} rating - Die Bewertung des Events, ausgedrückt durch den RatingEnum.
     */
    rating: RatingEnum;

    /**
     * @property {string} message - Die detaillierte Nachricht oder der Kommentar des Benutzers zum Event.
     */
    message: string;

    /**
     * Erstellt eine Instanz von EventFeedBackDto.
     * @param {string} userId - Die eindeutige Kennung des Benutzers.
     * @param {string} userName - Der Name des Benutzers.
     * @param {RatingEnum} rating - Die Bewertung des Events.
     * @param {string} message - Die Nachricht oder der Kommentar des Benutzers.
     */
    constructor(
        userId: string,
        userName: string,
        rating: RatingEnum,
        message: string
    ) {
        this.userId = userId;
        this.userName = userName;
        this.rating = rating;
        this.message = message;
    }
}

File: kids-app/apps/backend/src/main/java/com/example/demo/share/dataObjects/dtos/EventFeedBackDto.java
package com.example.demo.share.dataObjects.dtos;

import com.example.demo.share.dataObjects.enums.RatingEnum;

/**
 * @file EventFeedBackDto.java
 * @description Diese Klasse repräsentiert ein Datenübertragungsobjekt (DTO) für Event-Feedback.
 * Sie wird verwendet, um Feedback-Informationen zu einem Event zu kapseln,
 * einschließlich der Benutzer-ID, des Benutzernamens, einer Bewertung und einer Nachricht.
 */

/**
 * @class EventFeedBackDto
 * @description POJO (Plain Old Java Object) zur Speicherung von Event-Feedback-Daten.
 * Es enthält Felder für den Benutzer, der das Feedback gegeben hat, die Bewertung
 * und eine zugehörige Nachricht.
 */
public class EventFeedBackDto {
    /**
     * @private
     * @property {String} userId - Die ID des Benutzers, der das Feedback gegeben hat.
     */
    private String userId;
    /**
     * @private
     * @property {String} userName - Der Name des Benutzers, der das Feedback gegeben hat.
     */
    private String userName;
    /**
     * @private
     * @property {RatingEnum} rating - Die Bewertung des Events, definiert durch die `RatingEnum`.
     */
    private RatingEnum rating;
    /**
     * @private
     * @property {String} message - Die Feedback-Nachricht oder der Kommentar.
     */
    private String message;

    /**
     * @constructor
     * @description Standardkonstruktor für die `EventFeedBackDto`-Klasse.
     * Ermöglicht die Instanziierung ohne anfängliche Werte.
     */
    public EventFeedBackDto() {}

    // Getter und Setter

    /**
     * @method getUserId
     * @description Gibt die Benutzer-ID des Feedback-Gebers zurück.
     * @returns {String} Die Benutzer-ID.
     */
    public String getUserId() { return userId; }
    /**
     * @method setUserId
     * @description Setzt die Benutzer-ID des Feedback-Gebers.
     * @param {String} userId - Die neue Benutzer-ID.
     */
    public void setUserId(String userId) { this.userId = userId; }

    /**
     * @method getUserName
     * @description Gibt den Benutzernamen des Feedback-Gebers zurück.
     * @returns {String} Der Benutzername.
     */
    public String getUserName() { return userName; }
    /**
     * @method setUserName
     * @description Setzt den Benutzernamen des Feedback-Gebers.
     * @param {String} userName - Der neue Benutzername.
     */
    public void setUserName(String userName) { this.userName = userName; }

    /**
     * @method getRating
     * @description Gibt die Bewertung des Events zurück.
     * @returns {RatingEnum} Die Bewertung.
     */
    public RatingEnum getRating() { return rating; }
    /**
     * @method setRating
     * @description Setzt die Bewertung des Events.
     * @param {RatingEnum} rating - Die neue Bewertung.
     */
    public void setRating(RatingEnum rating) { this.rating = rating; }

    /**
     * @method getMessage
     * @description Gibt die Feedback-Nachricht zurück.
     * @returns {String} Die Nachricht.
     */
    public String getMessage() { return message; }
    /**
     * @method setMessage
     * @description Setzt die Feedback-Nachricht.
     * @param {String} message - Die neue Nachricht.
     */
    public void setMessage(String message) { this.message = message; }
}
